infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/protostream/ProtostreamPayload.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispanremote.impl.protostream;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+
+import org.hibernate.AssertionFailure;
+import org.hibernate.ogm.datastore.infinispanremote.impl.VersionedTuple;
+import org.hibernate.ogm.datastore.map.impl.MapTupleSnapshot;
+import org.hibernate.ogm.model.key.spi.AssociationKey;
+import org.hibernate.ogm.model.key.spi.RowKey;
+import org.hibernate.ogm.model.spi.Tuple;
+
+public final class ProtostreamPayload {
+
+	//One and only one of the following fields will be initialized:
+	private final MapTupleSnapshot loadedSnapshot;
+	private final Tuple tuple;
+
+	public ProtostreamPayload(MapTupleSnapshot loadedSnapshot) {
+		this.loadedSnapshot = Objects.requireNonNull( loadedSnapshot );
+		this.tuple = null;
+	}
+
+	public ProtostreamPayload(Tuple tuple) {
+		this.tuple = Objects.requireNonNull( tuple );
+		this.loadedSnapshot = null;
+	}
+
+	public void forEach(BiConsumer<String, Object> actionOnNamedTouples) {
+		if ( loadedSnapshot != null ) {
+			loadedSnapshot.getMap().forEach( actionOnNamedTouples );
```

Comment:
what does that do? Update the snapshot? Snapshots are supposed to be immutable I think.